vault.go

```diff
@@ -0,0 +1,234 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+
+	"github.com/blang/vfs"
+	vaultapi "github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/helper/jsonutil"
+)
+
+// logFatal is defined so log.Fatal calls can be overridden for testing
+var logFatal = log.Fatal
+
+// Vault -
+type Vault struct {
+	client *vaultapi.Client
+	fs     vfs.Filesystem
+}
+
+// NewVault - instantiate a new
+func NewVault() *Vault {
+	vaultConfig := vaultapi.DefaultConfig()
+
+	err := vaultConfig.ReadEnvironment()
+	if err != nil {
+		logFatal("Vault setup failed", err)
+	}
+
+	client, err := vaultapi.NewClient(vaultConfig)
+	if err != nil {
+		logFatal("Vault setup failed", err)
+	}
+
+	return &Vault{client, nil}
+}
+
+// Login -
+func (v *Vault) Login() {
+	v.client.SetToken(v.GetToken())
+}
+
+// GetToken -
+func (v *Vault) GetToken(fsOverrides ...vfs.Filesystem) string {
+	if len(fsOverrides) == 0 {
+		v.fs = vfs.OS()
+	} else {
+		v.fs = fsOverrides[0]
+	}
+
+	if token := v.AppRoleLogin(); token != "" {
+		return token
+	}
+	if token := v.AppIDLogin(); token != "" {
+		return token
+	}
+	if token := v.GitHubLogin(); token != "" {
+		return token
+	}
+	if token := v.UserPassLogin(); token != "" {
+		return token
+	}
+	if token := v.TokenLogin(); token != "" {
+		return token
+	}
+	logFatal("All vault auth failed")
+	return ""
+}
+
+// Logout -
+func (v *Vault) Logout() {
+}
+
+func (v *Vault) Read(path string) ([]byte, error) {
+	secret, err := v.client.Logical().Read(path)
+	if err != nil {
+		return nil, err
+	}
+
+	return jsonutil.EncodeJSON(secret.Data)
```

Comment:
I'd rather use the standard library method. I'll be migrating from using glide to dep at some point soon, and with dep I'd like to prune before building, so reducing dependencies is helpful.